### PR TITLE
Add geolocation static filter

### PR DESCRIPTION
**Pull request by Delerme**

### CODE REVIEW

**Comment by Delerme** [comment]
The backend work isn't done but I figured I'd at least get this into review and possibly pushed to start QAing in the meantime

File: README.md [hunk 1]
```diff
@@ -402,6 +402,30 @@ ANSWERS.addComponent('DateRangeFilter', {
...
+  // Radius around the user, in miles, to find results
+  radius: 50,
+  // The label to show when disabled (initial state)
+  disabledLabel: 'Use My Location',
+  // The label to show when enabled
+  enabledLabel: 'Disable My Location',
+  // The label to show when unable to get the user's location
+  errorLabel: 'Unable To Use Location',
+  // CSS selector of the button
+  buttonSelector: '.js-yext-geolocation-button'
```
Comment (benmcginnis) [inline README.md]: Mind keeping the yxt namespace?

File: src/ui/components/filters/geolocationcomponent.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    this.query = this.getUrlParams().get(`${this.name}.query`) || '';
+
+    /**
+     * The filter string to use for the provided query
+     * Optionally provided
+     * @type {string}
+     */
+    this.filter = this.getUrlParams().get(`${this.name}.filter`) || '';
+
+    if (this.query && this.query.length > 0 && this.filter && this.filter.length > 0) {
```
Comment (benmcginnis) [inline src/ui/components/filters/geolocationcomponent.js:83]: This logic feels like something that should be centralized, and this component would just ask that central manager for the current state of the filter for this component

File: src/ui/components/filters/geolocationcomponent.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  }
+
+  static defaultTemplateName () {
+    return 'controls/geolocation';
+  }
+
+  setState (data) {
+    super.setState({
+      ...data,
+      title: this._config.title,
```
Comment (benmcginnis) [inline src/ui/components/filters/geolocationcomponent.js:100]: Philosophically speaking, where are we drawing the line around setting config on the view model for a component and having the view access config directly?  I definitely prefer flattening the view model like this.

File: src/ui/components/filters/geolocationcomponent.js [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    });
+  }
+
+  onMount () {
+    if (this._autocomplete) {
+      this._autocomplete.remove();
+    }
+
+    this._initAutoComplete(this.inputSelector);
+    DOM.on(this._config.buttonSelector, 'click', () => this._toggleGeoFilter());
```
Comment (benmcginnis) [inline src/ui/components/filters/geolocationcomponent.js:115]: As compared to using config directly here, for example

File: src/ui/templates/controls/geolocation.hbs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+          {{#if geoButtonIcon}}
+            <img src="{{geoButtonIcon}}" class="yxt-GeoLocationFilter-buttonImage" alt="Use My Location">
+          {{/if}}
+          <span class="yxt-GeoLocationFilter-buttonText {{#if geoButtonIcon}}sr-only{{/if}}">
+            {{geoButtonText}}
+          </span>
+        </button>
+      </form>
+    </div>
+    <div class="js-yext-geo-autocomplete yxt-GeoLocationFilter-autocomplete"></div>
```
Comment (benmcginnis) [inline src/ui/templates/controls/geolocation.hbs]: might as well use the same class, but with `js-` appended for the js hook as the styling, I feel like that’s clearer

File: README.md (hunk 1 above)
Comment (Delerme) [reply README.md]: done

File: src/ui/components/filters/geolocationcomponent.js (hunk 2 above)
Comment (Delerme) [reply src/ui/components/filters/geolocationcomponent.js:83]: Yup that'll be part of a larger refactor that's already scheduled, just not prioritized

File: src/ui/components/filters/geolocationcomponent.js (hunk 3 above)
Comment (Delerme) [reply src/ui/components/filters/geolocationcomponent.js:100]: I think it's the wild west for the meantime, but I'd personally prefer a flat view model

File: src/ui/components/filters/geolocationcomponent.js (hunk 4 above)
Comment (Delerme) [reply src/ui/components/filters/geolocationcomponent.js:115]: yea wild wild west

File: src/ui/templates/controls/geolocation.hbs (hunk 5 above)
Comment (Delerme) [reply src/ui/templates/controls/geolocation.hbs]: done

**Review by benmcginnis — APPROVED** [review]

**Pull request merged**